Marlin/src/inc/SanityCheck.h

```diff
@@ -2873,6 +2873,29 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
   #endif
 #endif
 
+/**
+ * Screen Timeout feature not supported by these common displays
+*/
+#if HAS_SCREEN_TIMEOUT
+  #if HAS_MARLINUI_HD44780
+    #error "SCREEN_TIMEOUT feature not supported by HD44780 Reprap Discount Smart Controller 2004 LCD"
```

Comment:
Rather than checking HD44780 only
I think checking DOGLCD is better
`DOGLCD                  : Run a Graphical LCD through U8GLib (with MarlinUI)`